runtime/src/main/java/org/corfudb/runtime/view/BatchSequencerView.java

```diff
@@ -0,0 +1,252 @@
+            MultimapBuilder.hashKeys().hashSetValues().build();
+
+        while (!runtime.isShutdown()) {
+            readOps.clear();
+            writeOps.clear();
+            conditionalWriteOps.clear();
+            abortedOps.clear();
+
+            readStreams.clear();
+            tokenRequests.clear();
+            conditionalTokenRequests.clear();
+            conflictMap.clear();
+
+            try {
+                s.tryAcquire(runtime.getParameters().getBatchSequencerRequestCount()
+                    , runtime.getParameters().getBatchSequencerTimeout().toNanos(),
+                    TimeUnit.NANOSECONDS);
+            } catch (InterruptedException ie) {
+                throw new UnrecoverableCorfuInterruptedError(ie);
+            }
+            BatchSequencerOperation op;
+            while ((op = queue.pollFirst()) != null) {
+                if (op.next) {
+                    if (op.info == null) {
+                        tokenRequests.add(op.streams);
+                        writeOps.add(op);
+                    } else {
+                        boolean isLocallyConflicted = false;
+                        for (Entry<UUID, Set<byte[]>> e : op.info.getConflictSet().entrySet()) {
+                            Set<ByteBuffer> conflictSet = e.getValue().stream()
+                                .map(ByteBuffer::wrap).collect(Collectors.toSet());
+                            Set<ByteBuffer> written = conflictMap.get(e.getKey());
+                            Set<ByteBuffer> conflicts = Sets.intersection(written, conflictSet);
+                            if (!conflicts.isEmpty()) {
+                                isLocallyConflicted = true;
+                                op.response = new TokenResponse(TokenType.TX_ABORT_CONFLICT,
+                                    conflicts.iterator().next().array(),
+                                    new Token(Address.ABORTED, 0), Collections.emptyMap());
+                                abortedOps.add(op);
+                                break;
+                            }
+                        }
+                        if (!isLocallyConflicted) {
+                            op.info.getWriteConflictParams().forEach((id, writeSet) -> {
+                                Set<ByteBuffer> writes = writeSet.stream()
+                                    .map(ByteBuffer::wrap).collect(Collectors.toSet());
+                                conflictMap.putAll(id, writes);
+                            });
+                            conditionalTokenRequests.add(op.info);
+                            conditionalWriteOps.add(op);
+                        }
+                    }
+                } else {
+                    // Batch reads
+                    readOps.add(op);
+                    readStreams.addAll(Arrays.asList(op.streams));
+                }
+            }
+
+            if (readOps.size() != 0 || writeOps.size() != 0 || conditionalWriteOps.size() != 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)